esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+## Conventions
+
+### Namespaces
+
+The scripting engine requires strict adherence to namespaces correctness. A frequent source of scripting errors is omitting namespaces where they're required or using an incorrect namespace or namespace prefix.
+
+Namespaces referenced in any of your xpaths must be defined in your scripts to match the XML being processed. One exception is that the BackendRequest entry point will automatically detect and define the standard soap namespace ( `xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"` ) if it is used in your source XML.
+
+When your source XML uses a namespace prefix as part of the tag name, you must also use it in your xpath referencing that node.
+
+If a node defines a namespace without a prefix, you must define that namespace with a prefix in the root of your script. Then you must use the prefix in any xpaths referencing that node. Any children of that node in the source XML must not be referred to with the prefix you defined in your script:
+
+*source XML:*
+```xml
+<script_context_section>
+    <request url="http://certstagingvip.hpcc.risk.regn.net:9876">
+      <content>
+        <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
+          <soap:Body>
+            <inquiry_services.log_service xmlns="urn:hpccsystems:ecl:inquiry_services.log_service">
+              <city>minneapolis</city>
+              <state>mn</state>
+              <zip>55419</zip>
+            </inquiry_services.log_service>
+          </soap:Body>
+        </soap:Envelope>
+      </content>
+    </request>
+</script_context_section>
+```
+If I want to refer to the the `city` node above, my script would look like this:
+
+```xml
+<BackendRequest name="TestScript"
+                xmlns:es="urn:hpcc:esdl:script"
+                source="script_context_section"
+                xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"
+                xmlns:insvc="urn:hpccsystems:ecl:inquiry_services.log_service">
+    <es:set-value target="city-value"
+                  select="request/soap:Envelope/soap:Body/insvc:inquiry_services.log_service/city"/>
+</BackendRequest>
+```
+
+*Provide some examples*
+
+### Quoting Strings
+
+String tokens must be quoted in xpath expressions. Our xpath expressions occur in XML attributes, which also must be quoted. However, the quotes used to delimit the attribute cannot be contained inside the xpath expression. Usually this is handled by using double-quotes for attributes and  apostrophes (single quotes), for quoting inside the attribute. For example:
+
+```xml
+<es:set-value target="name" value="'foobar'"/>
+```
+
+In some cases you may see XML or xpath parsing errors when using the single quotes depending on your workflow (_establish precisely when this occurs_). This may be caused by unescaped apostrophes, which you can remedy like this:
```

Comment:
I've never had to do this in any of my test cases.  It could be something you remember from when we were parsing with ptree?  Anyway, I hope we only do this when we know it doesn't work otherwise.  And once we know when if ever we need this we should either remove this from the doc or make it very clear.

Reply:
I think you're right that this is no longer an issue. I wasn't able to find a case where I could cause a parsing error. Removed from document.